### ISSUE TITLE
Strange bug with crafter not consuming items unless the previous input slot is fully empty

### ISSUE THREAD

**Issue by WebDragon** [McJty/RFTools#1434]
I've set up a tier 2 crafter with mystical agriculture inferium -> prudentium essence infusion, and so on up the line. This worked brilliantly in Sky Factory 3, but in FTB Unstable (RFTools version 1.12-6.13) I am seeing a weird thing happen where the recipes work, but .. 

the inferium converts to prudentium fine
prudentium won't convert to intermedium, unless the inferium slot in the input buffer is fully emptied
intermedium won't convert to superium, unless both the inferium slot and the prudentium slot in the input buffer are both fully emptied.. 

I'm piping them to external except for the infusion crystal, and having that stay in the buffer, and using Transfer nodes to pipe the results back into the input buffer so I can work my way up to supremium

observe in the attached screenshots the prudentium just sitting there with enough to make 2 intermedium until I pick up the leftover 3 inferium in the first slot.

![2017-10-30_20 25 22](https://user-images.githubusercontent.com/1612570/32202277-e7c8c310-bdb1-11e7-836a-80b327e34bd4.png)
![2017-10-30_20 25 38](https://user-images.githubusercontent.com/1612570/32202278-e7d9c99e-bdb1-11e7-8701-050771a17abc.png)

**Comment by McJty** [comment]
This is an old version of rftools. Could be that this is already fixed

**Comment by WebDragon** [comment]
It would be fantastic if this repo tracked and hosted releases as well; be easier to reference them.

I was using the latest FTB Unstable 1.12 and assumed that RFTools would have been pretty recent due to 1.12-ness :) I'll see if I can update the mod without unbalancing the pack and get back to you on this

**Comment by WebDragon** [comment]
![screenshot_2017-11-06_14-56-05](https://user-images.githubusercontent.com/1612570/32461197-10ba105c-c303-11e7-9eca-47f00f9b4799.png)
![screenshot_2017-11-06_14-55-49](https://user-images.githubusercontent.com/1612570/32461198-10c5f750-c303-11e7-92cd-3f569f24d1d0.png)
Similar issues with 1.12-7.13 

I expected the crafter to allow me to reserve *at least* one spot for a particular material (the five grades of inferium essence) and then I leave the two crystals loose (because since I can't ignore their state, if I make a spot for one, it won't stay in it and uses up another slot), and I expect any inferium overflow to also be consumed even if it is not in that first slot.

Unfortunately it is (see attached) neither consuming the inferium that hits the overflow area, but if any essence occupies the previous slot, the next level up will not be converted either.

**Comment by WebDragon** [comment]
So I have noticed one other thing -- it works perfectly when you use the master infusion crystal -- the problem only presents when using the lower tier infusion crystal.

I originally made the recipes with the fully filled basic infusion crystal (so that the recipe isn't tied to the state of the crystal), but now that I've switched it over to the master crystal, I no longer experience the above issue.

HTH in diagnosis.

**Comment by choekstr** [comment]
Came here to report this same thing.  It seems the very first spot is required for whatever the recipe looks for.  For instance, if there are 4 inferium in the first slot then the recipe fires.  If there is inferium in other slots it will not.  It doesn't seem to matter where the infusing crystal is, just not the first slot as that won't work.

**Comment by WebDragon** [comment]
@choekstr try switching up to the Master Infusion Crystal -- that seems to work around the problem for now.

**Comment by choekstr** [comment]
yeah, switched gears and have been working for 2 days just to get the master infusion crystal.  Here is my recipe for the master infusion crystal.  Needless to say, it is a LOT of resources and time to get here, especially without the crafter working to make supremium essence. And man the mana required for 12 terrasteel ingots! [EOD:Synthesis recipe](https://user-images.githubusercontent.com/161085/34323146-4fc6fa4e-e801-11e7-8e5b-c4d090d47ef8.png) 

This crafter issue is definitely a 1.12.x bug as we all did this exact same thing in SF3 running 1.10.x and it worked perfectly.  Hoping @McJty has a chance to review the logic for the crafters and see what  has changed for 1.12.  The behavior is very odd indeed but there is a pattern and just requires moving the crystal around and placing the items to be crafted in the first slot.

**Comment by WebDragon** [comment]
In the meantine, my answer to that was to form a fresh crystal (the one that has durability loss) and remake all the recipes with that as the focus, and then just periodically wander by and hand-move around the essences into their respective slots until I had enough for the master crystal. a bit of work, but most early farms are anyway, and by the time I wandered back the farmer was full of seeds that could be also fed into the seed reprocessor for extra inferium. upgrade the seed tier as soon as you can, and let the farmer replant the whole 9x9 field from the extra seed drops, and toss all of the previous tier seeds into the reprocessor.
